programs/drift/src/state/user.rs

```diff
@@ -420,6 +431,142 @@ impl User {
 
         false
     }
+
+    pub fn increment_fuel_bonus(
+        &mut self,
+        fuel_bonus: u64,
+        user_stats: &mut UserStats,
+        now: i64,
+    ) -> DriftResult {
+        user_stats.update_fuel_bonus(fuel_bonus, now)?;
+        self.last_fuel_bonus_update_ts = now;
+
+        Ok(())
+    }
+
+    pub fn calculate_margin_and_increment_fuel_bonus(
+        &mut self,
+        perp_market_map: &PerpMarketMap,
+        spot_market_map: &SpotMarketMap,
+        oracle_map: &mut OracleMap,
+        context: MarginContext,
+        user_stats: &mut UserStats,
+        now: i64,
+    ) -> DriftResult<MarginCalculation> {
+        validate!(
+            context.fuel_bonus_numerator == user_stats.get_fuel_bonus_numerator(now)?,
```

Comment:
might save compute if you just check if it's set? idk

Reply:
makes sense, maybe just leaving this for now while change is in progress to ensure that things got updated properly